### PR TITLE
[Home] TidyBot 2.0

### PR DESCRIPTION
**Pull request by imgntn**
This PR adds tidy bot 2.0

- Triggered by running hand through 'beam'
- Triggered by mouse click
- Plays sound
- Flickers eyes
- Changes beam color
- Toggles light color

To test: go to forum, reload all running scripts, try tidy guy

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5442ffaf3eefee5ec8be5c9f687cc60b78cd4672 available at: https://gist.github.com/77ff2cd9c0a16ad786c66faa2e85bfc4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7881.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7881.dmg

**Comment by OmegaHeron** [comment]
Tested pushing its button - it does what PR advertises it should.  I glared at beam, willed my hands to move then remembered I don't have that tech.  PR pass for button pushing and action after button push.

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 1]
```diff
@@ -66,40 +66,107 @@
...
+                "Texture": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/Tidy-Swipe-Sticker.jpg",
+                "Texture.001": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/Head-Housing-Texture.png",
+                "Texture.003": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/Tidy-Swipe-Sticker-Emit.jpg",
+                "tex.button.blanks": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/Button-Blanks.png",
+                "tex.button.blanks.normal": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/Button-Blanks-Normal.png",
+                "tex.face.sceen.default": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/face-default.jpg",
+                "tex.face.screen-active": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/face-active.jpg",
+                "tex.glow.active": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/glow-active.png",
+                "tex.glow.default": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/glow-active.png"
             }
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 2]
```diff
@@ -66,40 +66,107 @@
...
 
+
             Entities.editEntity(_this.entityID, {
                 textures: JSON.stringify(data)
             });
         },
 
+        tidyGlowActive: function() {
+            print('tidy glow active')
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 3]
```diff
@@ -66,40 +66,107 @@
...
+            var res = Entities.findEntities(MyAvatar.position, 30);
+            var glow = null;
+            res.forEach(function(item) {
+                var props = Entities.getEntityProperties(item);
+                if (props.name === "Tidyguy-Glow") {
+                    glow = item;
+                }
+
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard: remove extra blank lines

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 4]
```diff
@@ -66,40 +66,107 @@
...
+
+            })
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 5]
```diff
@@ -66,40 +66,107 @@
...
+            if (glow !== null) {
+                print('editing glow' + glow)
+                Entities.editEntity(glow, {
+                    color: {
+                        red: 255,
+                        green: 140,
+                        blue: 20
+                    }
+                })
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 6]
```diff
@@ -66,40 +66,107 @@
...
+            }
+        },
+
+        tidyGlowDefault: function() {
+           print('tidy glow default')
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 7]
```diff
@@ -66,40 +66,107 @@
...
+            res.forEach(function(item) {
+                var props = Entities.getEntityProperties(item);
+                if (props.name === "Tidyguy-Glow") {
+                    glow = item;
+                }
+
+            });
+
+            if (glow !== null) {
+                print('editing glow' + glow)
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 8]
```diff
@@ -66,40 +66,107 @@
...
+           print('tidy glow default')
+
+            var res = Entities.findEntities(MyAvatar.position, 30);
+            var glow = null;
+            res.forEach(function(item) {
+                var props = Entities.getEntityProperties(item);
+                if (props.name === "Tidyguy-Glow") {
+                    glow = item;
+                }
+
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard: remove extra blank lines

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 9]
```diff
@@ -66,40 +66,107 @@
...
+            });
+
+            if (glow !== null) {
+                print('editing glow' + glow)
+
+                Entities.editEntity(glow, {
+
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard: remove extra blank lines

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 10]
```diff
@@ -66,40 +66,107 @@
...
+                    color: {
+                        red: 92,
+                        green: 255,
+                        blue: 209
+                    }
+                })
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 11]
```diff
@@ -66,40 +66,107 @@
...
+                "Texture": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/Tidy-Swipe-Sticker.jpg",
+                "Texture.001": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/Head-Housing-Texture.png",
+                "Texture.003": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/Tidy-Swipe-Sticker-Emit.jpg",
+                "tex.button.blanks": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/Button-Blanks.png",
+                "tex.button.blanks.normal": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/Button-Blanks-Normal.png",
+                "tex.face.sceen.default": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/face-default.jpg",
+                "tex.face.screen-active": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/face-active.jpg",
+                "tex.glow.active": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/glow-active.png",
+                "tex.glow.default": "atp:/Tidyguy-4d.fbx/Tidyguy-4d.fbm/glow-default.png"
             }
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 12]
```diff
@@ -109,6 +176,18 @@
         },
 
         playTidyingSound: function() {
+            var location = {
+                x: 1102.7660,
+                y: 460.9814,
+                z: -78.6451
+            }
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 13]
```diff
@@ -109,6 +176,18 @@
...
+
+            var injector = Audio.playSound(_this.tidySound, {
+                volume: 1,
+                position: location
+            });
+
+            print('INJECTOR IS: ' + injector)
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 14]
```diff
@@ -144,9 +277,9 @@
 
         },
 
-        startNearTrigger: function() {
-            _this.toggleButton();
-        },
+        // startNearTrigger: function() {
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard: remove dead code

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 15]
```diff
@@ -479,7 +612,28 @@
 
         },
 
+        update: function() {
+            if (_this.tidying === true) {
+                return
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard missing ;

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 16]
```diff
@@ -479,7 +612,28 @@
...
+            var leftHandPosition = MyAvatar.getLeftPalmPosition();
+            var rightHandPosition = MyAvatar.getRightPalmPosition();
+
+            var rightDistance = Vec3.distance(leftHandPosition, BEAM_POSITION)
+            var leftDistance = Vec3.distance(leftHandPosition, BEAM_POSITION)
+
+            if (rightDistance < BEAM_TRIGGER_THRESHOLD || leftDistance < BEAM_TRIGGER_THRESHOLD) {
+                _this.toggleButton();
+            } else {
+                //too far to toggle swipe
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard: remove dead code

File: unpublishedScripts/DomainContent/Home/reset.js [hunk 17]
```diff
@@ -479,7 +612,28 @@
...
+            }
+        },
+
         unload: function() {
+            Script.update.disconnect(_this.update);
+
+            Script.clearInterval(_this.flickerInterval);
             // this.findAndDeleteHomeEntities();
```
Comment (ZappoMan) [inline unpublishedScripts/DomainContent/Home/reset.js]: coding standard: remove dead code

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5d1b2996d3ab7a84100d592af8d61fbd43864bb8 available at: https://gist.github.com/77ff2cd9c0a16ad786c66faa2e85bfc4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7881.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7881.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0c5adf83ce3b8d45ad50944611c973e41d66512e available at: https://gist.github.com/77ff2cd9c0a16ad786c66faa2e85bfc4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7881.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7881.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 946558d0f6fb77942f3a14685d66e2c960f085ff available at: https://gist.github.com/77ff2cd9c0a16ad786c66faa2e85bfc4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7881.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7881.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d24ec08f9f69e394282a3956ff3383fa210d52a7 available at: https://gist.github.com/77ff2cd9c0a16ad786c66faa2e85bfc4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7881.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7881.dmg

**Pull request merged**